main.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import (
+	"os"
+
+	"github.com/bitrise-io/go-steputils/v2/stepconf"
+	"github.com/bitrise-io/go-utils/v2/env"
+	"github.com/bitrise-io/go-utils/v2/log"
+	"github.com/bitrise-io/go-utils/v2/pathutil"
+	"github.com/bitrise-steplib/steps-save-carthage-cache/step"
+)
+
+func main() {
+	os.Exit(run())
+}
+
+func run() int {
+	logger := log.NewLogger()
+	envRepo := env.NewRepository()
+	inputParser := stepconf.NewInputParser(envRepo)
+	pathChecker := pathutil.NewPathChecker()
+	pathProvider := pathutil.NewPathProvider()
+	pathModifier := pathutil.NewPathModifier()
+	cacheStep := step.New(logger, inputParser, pathChecker, pathProvider, pathModifier, envRepo)
+
+	exitCode := 0
+
+	if err := cacheStep.Run(); err != nil {
+		logger.Errorf(err.Error())
+		exitCode = 1
+		return exitCode
+	}
+
+	return exitCode
```

Comment:
Do you need the `exitCode` variable? At first I did not see how it was used and was reading this code piece multiple times to understand why it is needed. And it is not needed if I am right. 😄

This step code is simple so in my opinion 
```
if err := cacheStep.Run(); err != nil {
 	logger.Errorf(err.Error())
 	return 1
 }

 return 0
```
would make it clearer. Or maybe add some constants to have named variables if we really want to have them named.